ffc/client.py

```diff
@@ -165,7 +188,139 @@ def _is_token_expired(self):
             return True
 
 
+class HttpxFFCAPIClient:
+    """
+    HTTPX API Client
+    """
+
+    def __init__(self, base_url: str, sub, secret):
+        self.api_base_url = base_url
+        self._sub = sub
+        self._secret = secret
+        self._jwt = None
+        self.client = httpx.AsyncClient(base_url=self.api_base_url)
+
+    async def get_generated_charge_files(self) -> list[dict[str, Any]] | None:
+        """
+        This method fetches the charges files in the GENERATED Status.
+        This status means that the charges files have been generated but
+        not yet processed by the billing procedure.
+        Raises:
+        """
+        try:
+            headers = self._get_headers()
+            rql_filter = "eq(status,generated)"
+            endpoint = f"{self.api_base_url}/ops/v1/charges?{rql_filter}"
+            response = await self.client.get(headers=headers, url=endpoint)
+            response.raise_for_status()
+            data =  response.json()
+            return data.get("items", [])
+        except httpx.HTTPError as error:
+            logger.exception(f"Request to get charge files failed: {error}")
+            return None
+
+    async def get_charges_file_download_url(
+        self, charge_file: dict[str, Any], tmp_dir: str
+    ) -> str | None:
+        """
+        This method downloads asynchronously downloads a charge file from the FFC_OPERATIONS API
+        and stores it as a zip file in a temporary directory.
+        Args:
+            client (httpx.AsyncClient): The HTTP client used to make asynchronous requests.
+            charge_file (dict[str, Any]): A dictionary containing the charge file.
+            tmp_dir: The temporary directory to store the downloaded zip file.
+        """
+        charge_file_id = charge_file["id"]
+        initial_url = f"{self.api_base_url}/ops/v1/charges/{charge_file_id}/download"
+        headers = self._get_headers()
+        try:
+            response = await self.client.get(initial_url, headers=headers)
+            if response.status_code != 307:
+                logger.error(
+                    f"Unexpected status code {response.status_code} for {charge_file_id}"
+                )
+                return None
+            redirect_url = response.headers[
+                "Location"
+            ]  # extract the download url location from the headers
+            url_response = await self.client.get(
+                redirect_url
+            )  # the URL has a short-lived sas token
+            if url_response.status_code != 200:
+                logger.error(
+                    f"Failed to download  {charge_file_id}, status: {url_response.status_code}"
+                )
+                return None
+            file_path = os.path.join(tmp_dir, f"{charge_file_id}.zip")
+            return await read_the_response_in_chunks_and_write_to_file(
```

Comment:
this function is a perfect candidate to be a client "private" function, you don't use it anywhere else right ? So it doesn't make sense to have it outside the client.